pkg/cfn/builder/iam.go

```diff
@@ -272,7 +273,7 @@ func newIAMRoleResourceSet(name, namespace, serviceAccount, permissionsBoundary
 func (*IAMRoleResourceSet) WithIAM() bool { return true }
 
 // WithNamedIAM returns false
-func (rs *IAMRoleResourceSet) WithNamedIAM() bool { return rs.roleName != "" }
+func (rs *IAMRoleResourceSet) WithNamedIAM() bool { return true }
```

Comment:
Previously, this was fine, because on each update, it would just not care and not provide this so it the update didn't need this. And when create was defining it we added it. However, now, update sets rolename even if create did not so it always needs named IAM access.